separator.go

```diff
@@ -16,284 +16,29 @@
-		if i >= len(s.str) {
-			s.str = s.str[len(s.str):]
-			return
-		}
-
-		for ; i < len(s.str); i++ {
-			if l := len(terminate); l == 1 {
-				if string(s.str[i]) == terminate {
-					s.str = s.str[i+1:]
-					i = 0
-					break
-				}
-			} else if l == 2 {
-				if i+1 < len(s.str) && string(s.str[i:i+2]) == terminate {
-					s.str = s.str[i+2:]
-					i = 0
-					break
-				}
-			}
-		}
-
-		// not terminated, but end of string
-		if i >= len(s.str) {
-			s.str = s.str[len(s.str):]
-			return
-		}
-	}
-}
-
-// separate separates input string into multiple Spanner statements.
-// This does not validate syntax of statements.
-//
-// NOTE: Logic for parsing a statement is mostly taken from spansql.
-// https://github.com/googleapis/google-cloud-go/blob/master/spanner/spansql/parser.go
-func (s *separator) separate() []inputStatement {
-	var statements []inputStatement
-	for len(s.str) > 0 {
-		s.skipComments()
-		if len(s.str) == 0 {
-			break
-		}
-
-		switch s.str[0] {
-		// possibly string literal
-		case '"', '\'', 'r', 'R', 'b', 'B':
-			// valid string prefix: "b", "B", "r", "R", "br", "bR", "Br", "BR"
-			// https://cloud.google.com/spanner/docs/lexical#string_and_bytes_literals
-			raw, bytes, str := false, false, false
-			for i := 0; i < 3 && i < len(s.str); i++ {
-				switch {
-				case !raw && (s.str[i] == 'r' || s.str[i] == 'R'):
-					raw = true
-					continue
-				case !bytes && (s.str[i] == 'b' || s.str[i] == 'B'):
-					bytes = true
-					continue
-				case s.str[i] == '"' || s.str[i] == '\'':
-					str = true
-					switch {
-					case raw && bytes:
-						s.consumeRawBytesString()
-					case raw:
-						s.consumeRawString()
-					case bytes:
-						s.consumeBytesString()
-					default:
-						s.consumeString()
-					}
-				}
-				break
-			}
-			if !str {
-				s.sb.WriteRune(s.str[0])
-				s.str = s.str[1:]
-			}
-		// quoted identifier
-		case '`':
-			s.sb.WriteRune(s.str[0])
-			s.str = s.str[1:]
-			s.consumeStringContent("`", false)
-		// horizontal delim
-		case ';':
-			statements = append(statements, inputStatement{
-				statement: strings.TrimSpace(s.sb.String()),
-				delim:     delimiterHorizontal,
-			})
-			s.sb.Reset()
-			s.str = s.str[1:]
-		// possibly vertical delim
-		case '\\':
-			if len(s.str) >= 2 && s.str[1] == 'G' {
-				statements = append(statements, inputStatement{
-					statement: strings.TrimSpace(s.sb.String()),
-					delim:     delimiterVertical,
-				})
-				s.sb.Reset()
-				s.str = s.str[2:]
-				continue
-			}
-			s.sb.WriteRune(s.str[0])
-			s.str = s.str[1:]
-		default:
-			s.sb.WriteRune(s.str[0])
-			s.str = s.str[1:]
-		}
-	}
-
-	// flush remained
-	if s.sb.Len() > 0 {
-		if str := strings.TrimSpace(s.sb.String()); len(str) > 0 {
-			statements = append(statements, inputStatement{
-				statement: str,
-				delim:     delimiterUndefined,
-			})
-			s.sb.Reset()
-		}
+	var result []inputStatement
+	for _, stmt := range gsqlsep.SeparateInputPreserveComments(input, `\G`) {
+		stmt := stmt
```

Comment:
What is the reason we need to capture this in a local variable?

Reply:
I don't think it is needed.